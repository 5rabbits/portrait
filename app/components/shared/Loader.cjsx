React = require('react')

@Loader = React.createClass
  render: ->
    return (
      <div className="empty-view">
        <div className="empty-container">
          <div className="empty-dialog">
            <i className="fa fa-circle-o-notch fa-5x color-soft fa-spin "></i>
          </div>
        </div>
      </div>
    )

module.exports = @Loader
