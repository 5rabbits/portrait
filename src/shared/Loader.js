import React, { PureComponent } from 'react';

export default class Loader extends PureComponent {
  render() {
    return (
      <div className="empty-view">
        <div className="empty-container">
          <div className="empty-dialog">
            <i className="fa fa-circle-o-notch fa-5x color-soft fa-spin "></i>
          </div>
        </div>
      </div>
    );
  }
}
