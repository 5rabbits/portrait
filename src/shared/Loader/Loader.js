import React from 'react'

const Loader = () =>
  <div className="empty-view">
    <div className="empty-container">
      <div className="empty-dialog">
        <i className="fa fa-circle-o-notch fa-5x color-soft fa-spin" />
      </div>
    </div>
  </div>

export default Loader
