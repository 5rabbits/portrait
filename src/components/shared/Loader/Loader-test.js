import { React, toJSON, mount } from 'test/helper'
import Loader from 'shared/Loader'

describe('Loader', () => {
  it('should render with minimum props', () => {
    const component = mount(<Loader />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
