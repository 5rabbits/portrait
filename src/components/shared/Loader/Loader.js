import React from 'react'
import { EmptyView } from 'shared'

const Loader = () =>
  <EmptyView>
    <i className="fa fa-circle-o-notch fa-5x color-soft fa-spin" />
  </EmptyView>

export default Loader
