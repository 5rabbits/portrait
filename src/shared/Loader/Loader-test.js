import { React, expect, mount } from 'test/helper'
import Loader from 'shared/Loader'

describe('Loader', () => {
  it('should render with minimum props', () => {
    const component = mount(<Loader />)

    expect(component).to.be.ok
  })
})
