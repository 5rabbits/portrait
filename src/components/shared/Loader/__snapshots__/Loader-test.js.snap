// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render with minimum props 1`] = `
<Loader>
  <EmptyView>
    <div
      className="empty-view"
    >
      <div
        className="empty-container"
      >
        <div
          className="empty-dialog"
        >
          <i
            className="fa fa-circle-o-notch fa-5x color-soft fa-spin"
          />
        </div>
      </div>
    </div>
  </EmptyView>
</Loader>
`;
